python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1325,17 +1330,25 @@ def _construct_run_with_snapshots(
                     asset_key = output.properties.asset_key if output.properties else None
                     adjusted_output = output
 
-                    if (
-                        output.properties is not None
-                        and asset_key
-                        and asset_graph.has(asset_key)
-                        and output.properties.asset_execution_type is None
-                    ):
-                        adjusted_output = output._replace(
-                            properties=output.properties._replace(
-                                asset_execution_type=asset_graph.get(asset_key).execution_type
+                    if asset_key and asset_graph.has(asset_key):
+                        if partitions_definition is None:
+                            # this assumes that if one partitioned asset is in a run, all other partitioned
+                            # assets in the run have the same partitions definition.
+                            asset_node = asset_graph.get(asset_key)
+                            if isinstance(
+                                asset_node.partitions_def, TimeWindowPartitionsDefinition
```

Comment:
might be somewhat simpler to just save this isinstance check for down below when you actually construct the partitions subset object